src/main/java/fi/aalto/cs/apluscourses/utils/PluginResourceBundle.java

```diff
@@ -11,11 +19,46 @@ private PluginResourceBundle() {
 
   private static final ResourceBundle bundle = ResourceBundle.getBundle("resources");
 
+  private static final Map<Project, ResourceBundle> bundles = new HashMap<>();
```

Comment:
Should it be `Collections.synchronizedMap(new HashMap<>())`?

Reply:
Yes, that's a good idea